File: src/main/node/net/ecma/Library/Tests/Components/Navigation/Depth.mjs
'use strict';
var X3D = require('./x3d.mjs');
try {
	var fs = require('fs');
} catch (e) {
console.log("Problems loading fs. On browser?",e);
}
var SFString = require('./x3d.mjs');
var SFNode = require('./x3d.mjs');
var head = require('./x3d.mjs');
var MFNode = require('./x3d.mjs');
var meta = require('./x3d.mjs');
var Scene = require('./x3d.mjs');
var WorldInfo = require('./x3d.mjs');
var MetadataSet = require('./x3d.mjs');
var MetadataBoolean = require('./x3d.mjs');
var MFBool = require('./x3d.mjs');
var NavigationInfo = require('./x3d.mjs');
var MetadataString = require('./x3d.mjs');
var MFString = require('./x3d.mjs');
var MetadataDouble = require('./x3d.mjs');
var MFDouble = require('./x3d.mjs');
var Viewpoint = require('./x3d.mjs');
var Transform = require('./x3d.mjs');
var SFVec3f = require('./x3d.mjs');
var Shape = require('./x3d.mjs');
var Appearance = require('./x3d.mjs');
var Material = require('./x3d.mjs');
var Box = require('./x3d.mjs');
var X3D0 =  new X3D({

      profile : new SFString("Full"),
      version : new SFString("3.3"),
      head : new SFNode(
        new head({
          meta : new MFNode([
            new meta({
              name : new SFString("comment"),
              content : new SFString("World of Titania")}),

            new meta({
              name : new SFString("created"),
              content : new SFString("Wed, 16 Sep 2015 02:05:14 GMT")}),

            new meta({
              name : new SFString("creator"),
              content : new SFString("Holger Seelig")}),

            new meta({
              name : new SFString("generator"),
              content : new SFString("Titania V1.4.9, http://titania.create3000.de")}),

            new meta({
              name : new SFString("identifier"),
              content : new SFString("file:///home/holger/Projekte/Cobweb/excite/tests/Components/Navigation/Depth.x3d")}),

            new meta({
              name : new SFString("modified"),
              content : new SFString("Sun, 11 Sep 2016 05:53:00 GMT")})])})),
      Scene : new SFNode(
        new Scene({
          children : new MFNode([
            new WorldInfo({
              title : new SFString("Depth"),
              metadata : new MFNode([
                new MetadataSet({
                  name : new SFString("Titania"),
                  DEF : new SFString("Titania"),
                  reference : new SFString("http://titania.create3000.de"),
                  value : new SFNode(
                    new MetadataSet({
                      name : new SFString("AngleGrid"),
                      DEF : new SFString("AngleGrid"),
                      reference : new SFString("http://titania.create3000.de"),
                      value : new SFNode(
                        new MetadataBoolean({
                          name : new SFString("enabled"),
                          DEF : new SFString("enabled"),
                          reference : new SFString("http://titania.create3000.de"),
                          value : new MFBool([false])}))})),
                  value : new SFNode(
                    new MetadataSet({
                      name : new SFString("Grid"),
                      DEF : new SFString("Grid"),
                      reference : new SFString("http://titania.create3000.de"),
                      value : new SFNode(
                        new MetadataBoolean({
                          name : new SFString("enabled"),
                          DEF : new SFString("enabled_1"),
                          reference : new SFString("http://titania.create3000.de"),
                          value : new MFBool([true])}))})),
                  value : new SFNode(
                    new MetadataSet({
                      name : new SFString("Selection"),
                      DEF : new SFString("Selection"),
                      reference : new SFString("http://titania.create3000.de"),
                      value : new SFNode(
                        new MetadataSet({
                          name : new SFString("children"),
                          DEF : new SFString("children"),
                          reference : new SFString("http://titania.create3000.de"),
                          navigationInfo : new SFNode(
                            new NavigationInfo({
                              type : ["WALK","ANY"],
                              DEF : new SFString("_1")}))})),
                      value : new SFNode(
                        new MetadataSet({
                          name : new SFString("previous"),
                          DEF : new SFString("previous"),
                          reference : new SFString("http://titania.create3000.de")}))})),
                  value : new SFNode(
                    new MetadataSet({
                      name : new SFString("NavigationInfo"),
                      DEF : new SFString("NavigationInfo"),
                      reference : new SFString("http://titania.create3000.de"),
                      value : new SFNode(
                        new MetadataString({
                          name : new SFString("type"),
                          DEF : new SFString("type"),
                          reference : new SFString("http://titania.create3000.de"),
                          value : new MFString(["EXAMINE"])}))})),
                  value : new SFNode(
                    new MetadataSet({
                      name : new SFString("Viewpoint"),
                      DEF : new SFString("Viewpoint"),
                      reference : new SFString("http://titania.create3000.de"),
                      value : new SFNode(
                        new MetadataDouble({
                          name : new SFString("position"),
                          DEF : new SFString("position"),
                          reference : new SFString("http://titania.create3000.de"),
                          value : new MFDouble([3.7273065935322,3.12903272440423,8.73592237645378])})),
                      value : new SFNode(
                        new MetadataDouble({
                          name : new SFString("orientation"),
                          DEF : new SFString("orientation"),
                          reference : new SFString("http://titania.create3000.de"),
                          value : new MFDouble([-0.612643442740256,0.78037437123052,0.125235189918556,0.512384258680222])})),
                      value : new SFNode(
                        new MetadataDouble({
                          name : new SFString("centerOfRotation"),
                          DEF : new SFString("centerOfRotation"),
                          reference : new SFString("http://titania.create3000.de"),
                          value : new MFDouble([0,0,0])}))}))})])}),

            new Viewpoint({}),

            new Transform({
              DEF : new SFString("Box"),
              translation : new SFVec3f([0,0,-1]),
              scale : new SFVec3f([5,3,1]),
              children : new MFNode([
                new Shape({
                  appearance : new SFNode(
                    new Appearance({
                      material : new SFNode(
                        new Material({}))})),
                  geometry : new SFNode(
                    new Box({}))})])}),

            new NavigationInfo({
              USE : new SFString("_1")})])}))});
console.log(X3D0.toXMLNode());

File: src/main/c/net/coderextreme/Library/Tests/Components/Navigation/Depth.c
#include "C:/x3d-code/www.web3d.org/x3d/languages/c/Concretes.h"
void main(int argc, char ** argv) {
Browser browser = X3D.getBrowser();
X3D X3D0;
X3D0.profile = "Full";
X3D0.version = "3.3";
head head1 = createNode("head");
meta meta2 = createNode("meta");
meta2.name = "comment";
meta2.content = "World of Titania";
head1.meta = new MFNode();

head1.meta[0] = meta2;

meta meta3 = createNode("meta");
meta3.name = "created";
meta3.content = "Wed, 16 Sep 2015 02:05:14 GMT";
head1.meta[1] = meta3;

meta meta4 = createNode("meta");
meta4.name = "creator";
meta4.content = "Holger Seelig";
head1.meta[2] = meta4;

meta meta5 = createNode("meta");
meta5.name = "generator";
meta5.content = "Titania V1.4.9, http://titania.create3000.de";
head1.meta[3] = meta5;

meta meta6 = createNode("meta");
meta6.name = "identifier";
meta6.content = "file:///home/holger/Projekte/Cobweb/excite/tests/Components/Navigation/Depth.x3d";
head1.meta[4] = meta6;

meta meta7 = createNode("meta");
meta7.name = "modified";
meta7.content = "Sun, 11 Sep 2016 05:53:00 GMT";
head1.meta[5] = meta7;

head = head1;

WorldInfo WorldInfo9 = createNode("WorldInfo");
WorldInfo9.title = "Depth";
MetadataSet MetadataSet10 = createNode("MetadataSet");
MetadataSet10.name = "Titania";
MetadataSet10.DEF = "Titania";
MetadataSet10.reference = "http://titania.create3000.de";
MetadataSet MetadataSet11 = createNode("MetadataSet");
MetadataSet11.name = "AngleGrid";
MetadataSet11.DEF = "AngleGrid";
MetadataSet11.reference = "http://titania.create3000.de";
MetadataBoolean MetadataBoolean12 = createNode("MetadataBoolean");
MetadataBoolean12.name = "enabled";
MetadataBoolean12.DEF = "enabled";
MetadataBoolean12.reference = "http://titania.create3000.de";
MetadataBoolean12.value = new MFBool(new boolean[False]);
MetadataSet11.value = new MFNode();

MetadataSet11.value[0] = MetadataBoolean12;

MetadataSet10.value = new MFNode();

MetadataSet10.value[0] = MetadataSet11;

MetadataSet MetadataSet13 = createNode("MetadataSet");
MetadataSet13.name = "Grid";
MetadataSet13.DEF = "Grid";
MetadataSet13.reference = "http://titania.create3000.de";
MetadataBoolean MetadataBoolean14 = createNode("MetadataBoolean");
MetadataBoolean14.name = "enabled";
MetadataBoolean14.DEF = "enabled_1";
MetadataBoolean14.reference = "http://titania.create3000.de";
MetadataBoolean14.value = new MFBool(new boolean[True]);
MetadataSet13.value = new MFNode();

MetadataSet13.value[0] = MetadataBoolean14;

MetadataSet10.value[1] = MetadataSet13;

MetadataSet MetadataSet15 = createNode("MetadataSet");
MetadataSet15.name = "Selection";
MetadataSet15.DEF = "Selection";
MetadataSet15.reference = "http://titania.create3000.de";
MetadataSet MetadataSet16 = createNode("MetadataSet");
MetadataSet16.name = "children";
MetadataSet16.DEF = "children";
MetadataSet16.reference = "http://titania.create3000.de";
NavigationInfo NavigationInfo17 = createNode("NavigationInfo");
NavigationInfo17.DEF = "_1";
NavigationInfo17.type = new MFString(new java.lang.String[", ","WALKANY"]);
MetadataSet16.value = NavigationInfo17;

MetadataSet15.value = new MFNode();

MetadataSet15.value[0] = MetadataSet16;

MetadataSet MetadataSet18 = createNode("MetadataSet");
MetadataSet18.name = "previous";
MetadataSet18.DEF = "previous";
MetadataSet18.reference = "http://titania.create3000.de";
MetadataSet15.value[1] = MetadataSet18;

MetadataSet10.value[2] = MetadataSet15;

MetadataSet MetadataSet19 = createNode("MetadataSet");
MetadataSet19.name = "NavigationInfo";
MetadataSet19.DEF = "NavigationInfo";
MetadataSet19.reference = "http://titania.create3000.de";
MetadataString MetadataString20 = createNode("MetadataString");
MetadataString20.name = "type";
MetadataString20.DEF = "type";
MetadataString20.reference = "http://titania.create3000.de";
MetadataString20.value = new MFString(new java.lang.String["EXAMINE"]);
MetadataSet19.value = new MFNode();

MetadataSet19.value[0] = MetadataString20;

MetadataSet10.value[3] = MetadataSet19;

MetadataSet MetadataSet21 = createNode("MetadataSet");
MetadataSet21.name = "Viewpoint";
MetadataSet21.DEF = "Viewpoint";
MetadataSet21.reference = "http://titania.create3000.de";
MetadataDouble MetadataDouble22 = createNode("MetadataDouble");
MetadataDouble22.name = "position";
MetadataDouble22.DEF = "position";
MetadataDouble22.reference = "http://titania.create3000.de";
MetadataDouble22.value = new MFDouble(new double[3.7273065935322,3.12903272440423,8.73592237645378]);
MetadataSet21.value = new MFNode();

MetadataSet21.value[0] = MetadataDouble22;

MetadataDouble MetadataDouble23 = createNode("MetadataDouble");
MetadataDouble23.name = "orientation";
MetadataDouble23.DEF = "orientation";
MetadataDouble23.reference = "http://titania.create3000.de";
MetadataDouble23.value = new MFDouble(new double[-0.612643442740256,0.78037437123052,0.125235189918556,0.512384258680222]);
MetadataSet21.value[1] = MetadataDouble23;

MetadataDouble MetadataDouble24 = createNode("MetadataDouble");
MetadataDouble24.name = "centerOfRotation";
MetadataDouble24.DEF = "centerOfRotation";
MetadataDouble24.reference = "http://titania.create3000.de";
MetadataDouble24.value = new MFDouble(new double[0,0,0]);
MetadataSet21.value[2] = MetadataDouble24;

MetadataSet10.value[4] = MetadataSet21;

WorldInfo9.metadata = MetadataSet10;

children = new MFNode();

children[0] = WorldInfo9;

Viewpoint Viewpoint25 = createNode("Viewpoint");
children[1] = Viewpoint25;

Transform Transform26 = createNode("Transform");
Transform26.DEF = "Box";
Transform26.translation = new SFVec3f(new float[0,0,-1]);
Transform26.scale = new SFVec3f(new float[5,3,1]);
Shape Shape27 = createNode("Shape");
Appearance Appearance28 = createNode("Appearance");
Material Material29 = createNode("Material");
Appearance28.material = Material29;

Shape27.appearance = Appearance28;

Box Box30 = createNode("Box");
Shape27.geometry = Box30;

Transform26.child = new undefined();

Transform26.child[0] = Shape27;

children[2] = Transform26;

NavigationInfo NavigationInfo31 = createNode("NavigationInfo");
NavigationInfo31.USE = "_1";
children[3] = NavigationInfo31;

}
